### ISSUE TITLE
Unusable with Steam Deck Docked

### ISSUE THREAD

**Issue by LiamDawe** [CryoByte33/steam-deck-utilities#68]
Hey, so the app is currently unusable when you have a Steam Deck Docked, because the UI for the install and app suddenly becomes massive like it's confusing on resolution or DPI or something.

Can you please test with your Deck docked and see what happens?

**Comment by CryoByte33** [comment]
The UI is "large" scale when a second display is connected and "small" when only the internal screen is present by design, see #37 . Are you saying that it's a large scale with a dock connected, but no external display?

**Comment by ajite** [comment]
I think we can keep this issue open until we find a better solution. I went through Fyne docs and want to try something.

At the moment It's better than before but in some case it might get too big.

**Comment by LiamDawe** [comment]
> The UI is "large" scale when a second display is connected and "small" when only the internal screen is present by design, see #37 . Are you saying that it's a large scale with a dock connected, but no external display?

The problem is it becomes multiple times as large as it should be. Even when moving it over to my much bigger screen when docked, it’s many times too big. I’ll try to get a shot of it for you.

**Comment by gezawatt** [comment]
> 
I got you.
![huge_ui](https://user-images.githubusercontent.com/62294693/221165246-b55fe8c4-905e-4ef8-b919-0170ce6a6758.png)

This only happens when I move the window to a specific part of my internal/external screen.

**Comment by CryoByte33** [comment]
If you right click on the title bar and press "move to" to move it to another display does it get fixed?

**Comment by gezawatt** [comment]
@CryoByte33 oh, that's interesting
![error1](https://user-images.githubusercontent.com/62294693/221167014-63dd4aaf-1e68-480d-97c7-3401ca139aa2.png)

![error2](https://user-images.githubusercontent.com/62294693/221167531-bb50ee81-c2f8-4dfc-95c3-08a9d560d4f3.png)

It thinks it's still on the internal display, even if I'm clicking this option while the app is on my external display.

**Comment by gezawatt** [comment]
> Does toggling it back and forth between displays help?

Nope. If I toggle it to another screen in this state it just flies beyond the screen, where I can't see it  or click it.

I have some free time today so I might install Go development environment and play around with the app.
I'm thinking **maybe** it's a good idea to just have the app use a static UI scaling for now, with the manual "UI scaling"  button in the GUI. Though that would require storing user-chosen UI scaling option in a config file, so the app wouldn't be stateless anymore... Hmm

**Comment by CryoByte33** [comment]
> I'm thinking maybe it's a good idea to just have the app use a static UI scaling for now

Yeah, the issue is Fyne being a pain to scale, because "scaling" is relative to the "current" display. 1 on the Deck screen is huge and is fairly small on something 1080p or above.

It _may_ be in our best interest to open an issue with Fyne themselves, since this is clearly so difficult to work around. I'm busy moving for the weekend, but hopefully I can get around to it next week.

**Comment by novhack** [comment]
This scaling problem happens to me even when I don't have any other screen connected. I have no idea what the problem is since scale on the Deck screen is set to 800x1280 and 100% scale.

**Comment by CryoByte33** [comment]
@novhack The issue is the 100% scale. Since the vertical resolution of the Deck is 1280, Fyne misreads the scaling factor. You need to manually override with 0.25-0.3 scale when launching, which is the entire purpose of launcher.sh.
